torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -213,7 +213,7 @@ class TORCH_API ProcessGroup : public torch::CustomClassHolder {
   // to the dispatcher.
   // TODO: Find a way to force the above rule programmatically.
   virtual c10::intrusive_ptr<ProcessGroup::Work> broadcast(
-      std::vector<at::Tensor>& /* tensors */,
+      const std::vector<at::Tensor>& /* tensors */,
```

Comment:
hmm, there are external libraries inherit from these APIs. Might need to sync with them. 

https://github.com/openucx/torch-ucc
https://github.com/oneapi-src/oneCCL

Reply:
Yea, I think this change is risky given we have 3rd party adopters. I'm fine with leaving this pull request open, and Howard to take it over.